docs/ingestion-sink/index.md

```diff
@@ -0,0 +1,224 @@
+load jobs will be submitted by each running instance of ingestion-sink.
+
+If none of the above configuration options are provided, then messages must be
+notifications from `BIG_QUERY_OUTPUT_MODE=file_loads` or
+`BIG_QUERY_OUTPUT_MODE=mixed`, and the blobs they indicate will be submitted to
+BigQuery via load jobs.
+
+### BigQuery
+
+`OUTPUT_TABLE` must be a `tableSpec` of form `dataset.tablename`
+or the more verbose `projectId.dataset.tablename`. The values can contain
+attribute placeholders of form `${attribute_name}`. To set dataset to the
+document namespace and table name to the document type, specify:
+
+    OUTPUT_TABLE='${document_namespace}.${document_type}'
+
+All `-` characters in the attributes will be converted to `_` per BigQuery
+naming restrictions. Additionally, document namespace and type values will
+be processed to ensure they are in snake case format (`untrustedModules`
+becomes `untrusted_modules`).
+
+Defaults for the placeholders using `${attribute_name:-default_value}`
+are supported, but likely don't make much sense since it's unlikely that
+there is a default table whose schema is compatible with all potential
+payloads.
+
+### Attribute placeholders
+
+We support routing individual messages to different output locations based on
+the `PubsubMessage` attribute map. Routing is accomplished by adding
+placeholders of form `${attribute_name:-default_value}` to the path.
+
+For example, to route based on a `document_type` attribute, your path might
+look like:
+
+    OUTPUT_BUCKET=gs://mybucket/mydocs/${document_type:-UNSPECIFIED}/myfileprefix
+
+Messages with `document_type` of "main" would be grouped together and end up in
+the following directory:
+
+    gs://mybucket/mydocs/main/
+
+Messages with `document_type` set to `null` or missing that attribute
+completely would be grouped together and end up in directory:
+
+    gs://mybucket/mydocs/UNSPECIFIED/
+
+Note that placeholders _must_ specify a default value so that a poorly
+formatted message doesn't cause a pipeline exception. A placeholder without a
+default will result in an `IllegalArgumentException` on pipeline startup.
+
+File-based outputs support the additional _derived_ attributes
+`"submission_date"` and `"submission_hour"` which will be parsed from the value
+of the `submission_timestamp` attribute if it exists.  These can be useful for
+making sure your output specification buckets messages into hourly directories.
+
+The templating and default syntax used here is based on the
+[Apache commons-text `StringSubstitutor`](https://commons.apache.org/proper/commons-text/javadocs/api-release/org/apache/commons/text/StringSubstitutor.html),
+which in turn bases its syntax on common practice in bash and other Unix/Linux
+shells.  Beware the need for proper escaping on the command line (use `\$` in
+place of `$`), as your shell may try to substitute in values for your
+placeholders before they're passed to `Sink`.
```

Comment:
A full example of this may help prevent confusion, perhaps later in the section with maven? This definitely confused me when I first worked with the Java tooling.

```
#!/bin/bash
OUTPUT=gs://\${document_type:-UNSPECIFIED} mvn ...
```

Reply:
fixed via https://github.com/mozilla/gcp-ingestion/pull/1212#r461911016